File: app/src/test/java/email/schaal/ocreader/StringUtilsTest.java
package email.schaal.ocreader;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;

import java.util.Date;

import androidx.test.core.app.ApplicationProvider;
import androidx.test.ext.junit.runners.AndroidJUnit4;
import email.schaal.ocreader.util.StringUtils;

import static org.junit.Assert.assertEquals;

/**
 * Test class for StringUtils
 */
@RunWith(AndroidJUnit4.class)
public class StringUtilsTest {
    @Test
    public void testGetByLine() throws Exception {
        assertEquals("", StringUtils.getByLine(ApplicationProvider.getApplicationContext(), "<p class=\"byline\">%s</p>", null));
        assertEquals("<p class=\"byline\">by testAuthor</p>", StringUtils.getByLine(ApplicationProvider.getApplicationContext(), "<p class=\"byline\">%s</p>", "testAuthor"));
    }

    @Test
    public void testGetTimeSpanString() throws Exception {
        Date testDateStart = new Date(1469849100000L);

        Date testDateMinute = new Date(1469849100000L + 60 * 1000);
        Date testDateHour = new Date(1469849100000L + 60 * 60 * 1000);
        Date testDateDay = new Date(1469849100000L + 24 * 60 * 60 * 1000);

        assertEquals("now", StringUtils.getTimeSpanString(ApplicationProvider.getApplicationContext(), testDateStart, testDateStart));
        assertEquals("1m", StringUtils.getTimeSpanString(ApplicationProvider.getApplicationContext(), testDateStart, testDateMinute));
        assertEquals("1h", StringUtils.getTimeSpanString(ApplicationProvider.getApplicationContext(), testDateStart, testDateHour));
        assertEquals("1d", StringUtils.getTimeSpanString(ApplicationProvider.getApplicationContext(), testDateStart, testDateDay));
    }

    @Test
    public void testCleanString() throws Exception {
        String html = "<span>Test</span>";
        String entity = "Test &gt; Test";

        assertEquals("Test", StringUtils.cleanString(html));
        assertEquals("Test > Test", StringUtils.cleanString(entity));
    }
}

File: app/src/test/java/email/schaal/ocreader/StringUtilsTest.kt
package email.schaal.ocreader

import android.graphics.Color
import android.util.Pair
import androidx.test.core.app.ApplicationProvider
import androidx.test.ext.junit.runners.AndroidJUnit4
import email.schaal.ocreader.database.model.Item
import email.schaal.ocreader.util.asCssString
import email.schaal.ocreader.util.cleanString
import email.schaal.ocreader.util.getByLine
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import java.util.*

/**
 * Test class for StringUtils
 */
@RunWith(AndroidJUnit4::class)
class StringUtilsTest {
    @Test
    @Throws(Exception::class)
    fun testGetCssColor() {
        val colorPairs: MutableList<Pair<Int, String>> = ArrayList(3)
        colorPairs.add(Pair(Color.argb(30, 30, 30, 30), "rgba(30,30,30,0.12)"))
        colorPairs.add(Pair(Color.argb(0, 0, 0, 0), "rgba(0,0,0,0.00)"))
        colorPairs.add(Pair(Color.argb(255, 255, 255, 255), "rgba(255,255,255,1.00)"))
        for (colorPair in colorPairs) Assert.assertEquals(colorPair.second, colorPair.first.asCssString())
    }

    @Test
    @Throws(Exception::class)
    fun testGetByLine() {
        val item = Item()
        Assert.assertEquals("", getByLine(ApplicationProvider.getApplicationContext(), "<p class=\"byline\">%s</p>", null, item.feed))
        Assert.assertEquals("<p class=\"byline\">by testAuthor</p>", getByLine(ApplicationProvider.getApplicationContext(), "<p class=\"byline\">%s</p>", "testAuthor", item.feed))
    }

    @Test
    @Throws(Exception::class)
    fun testCleanString() {
        val html = "<span>Test</span>"
        val entity = "Test &gt; Test"
        Assert.assertEquals("Test", html.cleanString())
        Assert.assertEquals("Test > Test", entity.cleanString())
    }
}